### PR TITLE
Examples in README and documentation #65

### PR DESCRIPTION
**Pull request by clason**
This PR removes the obsolete (no longer working) example from the README and adds a link to the documentation, where the finite-difference example from #65 is added.

### CODE REVIEW

**Comment by dlfivefifty** [comment]
Since the READMe examples now work, I've altered this PR to leave them in.

**Comment by clason** [comment]
Yes, I was about to amend the PR to put them back in. Thanks! Let me know if the example for the documentation should be amended in some way.

File: docs/src/index.md [hunk 1]
```diff
@@ -42,8 +42,14 @@ julia> BandedMatrix(Ones(5,5), (1,2))
            1.0  1.0  1.0
                 1.0  1.0
 ```
-
-
+To create a banded matrix of a given size with constant bands (such as the classical finite difference approximation of the two-dimensional Laplacian on the unit square [0,1]×[0,1]), you can use the following:
+```julia
+n = 128
+h = 1/n
+A = BandedMatrix{Float64}(undef, (n*n,n*n), (1,1))
```
Comment (dlfivefifty) [inline docs/src/index.md]: This should be an `(n,n)` not `(n*n,n*n)`.

File: docs/src/index.md [hunk 2]
```diff
@@ -42,8 +42,14 @@ julia> BandedMatrix(Ones(5,5), (1,2))
            1.0  1.0  1.0
                 1.0  1.0
 ```
-
-
+To create a banded matrix of a given size with constant bands (such as the classical finite difference approximation of the two-dimensional Laplacian on the unit square [0,1]×[0,1]), you can use the following:
```
Comment (dlfivefifty) [inline docs/src/index.md]: The code below does the one-dimensional Laplacian on the the unit interval `[0,1]`.

File: docs/src/index.md (hunk 1 above)
Comment (clason) [reply docs/src/index.md]: Note that -- differently from your example -- I am constructing the *2D* Laplacian (note the `-4.0` on the diagonal) which has size n^2 for an n-by-n grid (with mesh size 1/n).

But if you prefer, I can switch back to the 1D Laplacian. (I just thought the 2D case is more often used.)

Comment (dlfivefifty) [reply docs/src/index.md]: The 2D Laplacian is not a tridiagonal matrix: its block tridiagonal with tridiagonal blocks.

It is more naturally represented as a BandedBlockBandedMatrix from BlockBandedMatrices.jl

File: docs/src/index.md (hunk 2 above)
Comment (clason) [reply docs/src/index.md]: Fair enough, without reordering afterwards this is not useful. I'll change it back.

**Comment by clason** [comment]
And, yes, I'm looking forward to a Block `kron` to construct the 2D Laplacian from this 😄

**Comment by dlfivefifty** [comment]
😊 That should come soon. (Including GPU backends.)

**Pull request merged**